File: app/src/main/java/com/purpletealabs/bitcoinapp/datasource/IPriceDataSource.java
package com.purpletealabs.bitcoinapp.datasource;

import com.purpletealabs.bitcoinapp.dtos.Price;

import java.util.List;

public interface IPriceDataSource {
    interface Callback {
        void onGetDefaultPriceResult(List<Price> prices);

        void onGetDefaultPriceFailure();

        void onGetCurrencyPriceResult(Price price);

        void onGetCurrencyPriceFailure();
    }

    void getDefaultPriceList(Callback callback);

    void getPriceForCurrency(String currencyCode, Callback callback);
}

File: app/src/main/java/com/purpletealabs/bitcoinapp/datasource/IPriceDataSource.kt
package com.purpletealabs.bitcoinapp.datasource

import com.purpletealabs.bitcoinapp.dtos.Price

interface IPriceDataSource {
    fun cancel()
    interface Callback {
        fun onGetDefaultPriceResult(prices: List<Price>)

        fun onGetDefaultPriceFailure()

        fun onGetCurrencyPriceResult(price: Price)

        fun onGetCurrencyPriceFailure()
    }

    fun getDefaultPriceList(callback: Callback)

    fun getPriceForCurrency(currencyCode: String, callback: Callback)
}
